BenchPress/Generators/AzureDeploymentImporter.cs

```diff
@@ -229,4 +213,139 @@ string outputFolderPath
+            s_squareBracketSubstituition
+        );
+
+        // Find all matches in the parameter/variable string that follows the pattern of "parameters('...')" or
+        // "variables('...')". The regular expression pattern defines two named subexpressions: paramOrVarType, which
+        // represents the type of parameter/variable (e.g., "parameters" or "variables"), and paramOrVarName, which
+        // represents the name of the parameter/variable.
+        var matches = s_parametersOrVariablesRegex.Matches(stringToResolve);
+        foreach (Match match in matches)
+        {
+            var name = match.Groups[s_paramOrVarNameGroupKey].Value;
+            var type = match.Groups[s_paramOrVarTypeGroupKey].Value;
+            var resolvedValue = System.String.Empty;
+
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                // ARM templates will contain a JSON Object representing parameters and variables for the ARM template.
+                // Get the correct JSON Object using the type from the regex match (e.g., "parameters" or "variables").
+                var parametersOrVariablesObj = armTemplateObject[type];
+                if (parametersOrVariablesObj != null)
+                {
+                    // Get the value of the parameter/variable from the JSON Object. If the value is still a JSON
+                    // Object, that means it is a parameter that has a default value, so get the default value (if it
+                    // exists). Otherwise, the value can be converted to a string and assigned to resolveValue.
+                    var resolvedValueNode = parametersOrVariablesObj[name];
+                    if (resolvedValueNode != null)
+                    {
+                        if (resolvedValueNode is JsonObject)
+                        {
+                            resolvedValue = resolvedValueNode[s_defaultValueKey]?.ToString();
+                        }
+                        else
+                        {
+                            resolvedValue = resolvedValueNode.ToString();
+                        }
+                    }
+                }
+            }
+            if (!string.IsNullOrWhiteSpace(resolvedValue))
+            {
+                // Find and remove square brackets from the resolved value. Square brackets are specific to ARM
+                // template syntax and are not needed in generated tests.
+                resolvedValue = Regex.Replace(
+                    resolvedValue,
+                    s_squareBracketPattern,
+                    s_squareBracketSubstituition
+                );
+                // If the resolved value does not contain addtional ARM template functions, then wrap it in single
+                // quotes because the resolved value is a plain string type.
+                if (!s_parenthesesRegex.IsMatch(resolvedValue))
+                {
+                    resolvedValue = "\'" + resolvedValue + "\'";
+                }

```

Comment:
In the case of a value, would all ARM template functions be wrapped in square brackets?

If so, then would it be better to test for square brackets, and if they exist remove them, and if not then wrap in single quotes? It would remove the need for the parentheses regex.

If not, then I guess this is probably the best approach.

Reply:
Wow totally missed that the replacing square brackets can be reused to check for ARM template functions🤦‍♀️ you're a genius! LOL 

Change incoming!

Reply:
Alright, pushed the changes. Let me know if there's any follow up concerns!